### PR TITLE
test(client): port count test to functional

### PR DESCRIPTION
**Pull request by danstarns**
Replaces the existing integration count test with the new functional style.

### CODE REVIEW

File: packages/client/src/__tests__/integration/happy/count/test.ts [hunk 1]
```diff
@@ -2,7 +2,8 @@ import { getTestClient } from '../../../../utils/getTestClient'
 
 let PrismaClient, prisma
 
-describe('count', () => {
+// TODO - Delete me. Replaced in https://github.com/prisma/prisma/pull/13198
+describe.skip('count', () => {
```
Comment (aqrln) [inline packages/client/src/__tests__/integration/happy/count/test.ts:6]: Should we maybe remove it in this PR?

File: packages/client/tests/functional/count/prisma/_schema.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+export default ({ provider, id, foreignKeyId }) => {
```
Comment (SevInf) [inline packages/client/tests/functional/count/prisma/_schema.ts:1]: You can also `testMatrix.setupSchema` here to get typechecking for parameters

File: packages/client/tests/functional/count/_matrix.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { defineMatrix } from '../_utils/defineMatrix'
+
+export default defineMatrix(() => [
+  [
+    {
+      provider: 'sqlite',
+      id: 'Int @id @default(autoincrement())',
+      foreignKeyId: 'Int?',
+      providerFeatures: '',
```
Comment (aqrln) [inline packages/client/tests/functional/count/_matrix.ts:9]: I don't think you use `providerFeatures` in the schema, should we remove them from the matrix?

**Review by SevInf — APPROVED** [review]

File: packages/client/tests/functional/count/tests.ts [hunk 4]
```diff
@@ -35,7 +30,7 @@ describe('count', () => {
     const value = await prisma.user.count({
       select: true,
       where: {
-        age: 84,
+        age: 111,
```
Comment (millsp) [inline packages/client/tests/functional/count/tests.ts:33]: That's old 🤣

Comment (danstarns) [reply packages/client/tests/functional/count/tests.ts:33]: 🕸

**Pull request merged**